src/github.com/matrix-org/dendrite/clientapi/routing/filter.go

```diff
@@ -111,13 +111,13 @@ func PutFilter(
 		}
 	}
 
-	filterID, err := accountDB.PutFilter(req.Context(), localpart, string(filterArray))
+	filterID, err := accountDB.PutFilter(req.Context(), localpart, filterArray)
 	if err != nil {
 		return httputil.LogThenError(req, err)
 	}
 
 	return util.JSONResponse{
 		Code: 200,
-		JSON: filterResponse{FilterID: filterID},
+		JSON: filterResponse{FilterID: string(filterID)},
```

Comment:
This looks to be due to confusion between if PutFilter returns the filter json or the filterID